File: Hello_2020/Problem2.java
package Hello_2020;

import java.util.ArrayList;
import java.util.Scanner;
public class Problem2 {
  public static void main(String[] args) {
    Scanner s = new Scanner(System.in);
    int n = s.nextInt();
    int ascentCount  =  0;
    ArrayList<int[]> startEnd = new ArrayList<int[]>();
    int count = 0;
    for(int a = 0; a < n; a ++){
      int size = s.nextInt();
      boolean ascent = false;
      int[] values = new int[size];
      for(int b = 0; b < size; b++){
        values[b] = s.nextInt();
      }
      for(int b = 0; b < size-1; b++){
        if(values[b] < values[b+1]){
          ascent = true;
          ascentCount++;
          break;
        }
      }
      if(ascent == false){
        int[] vals = new int[2];
        vals[0] = values[0];
        vals[1] = values[values.length-1];
        startEnd.add(vals);
      }
    }
    count += ascentCount*n;
    count += (n-ascentCount) * ascentCount;
    for(int[] a:startEnd){
      for(int[] b: startEnd){
        if(a[1] < b[0]){
          count++;
        }
      }
    }
    System.out.println(count);
    s.close();
  }
}

File: Hello_2020/Problem2.py
n = int(input())
ascentCount = 0
startEnd = []
count = 0
for a in range(n):
    values = list(map(int,input().split()[1:]))
    ascent = False
    for b in range(len(values)-1):
        if values[b] < values[b+1]:
            ascentCount += 1
            ascent = True
            break
    if ascent == False:
        vals = []
        vals.append(values[0])
        vals.append(values[-1])
        startEnd.append(vals)
count += ascentCount * n
count += (n-ascentCount) * ascentCount
for a in startEnd:
    for b in startEnd:
        if a[-1] < b[0]:
            count += 1
print(count)
